### PR TITLE
[MER-503] Invited authors can confirm email and resend confirmation link

### PR DESCRIPTION
**Pull request by gastonabella**
[MER-503](https://eliterate.atlassian.net/browse/MER-503)

### Description
There are different states where an author could be depending on if it was invited by an admin or if it created an account by himself. These states are translated into different combinations of the following author fields added by the Pow library: `email_confirmation_token`, `email_confirmed_at`, `unconfirmed_email`, `invitation_token`, `invitation_accepted_at`.

There are some cases where an author is not waiting for email confirmation, and the author details view still allows a sys admin to Resend a confirmation link or to Confirm an author email manually, causing an Internal Server Error when attempting to perform these actions.

**Note**:
The original description in the Jira ticket is not a bug. It is a decision of Pow devs not to send a confirmation link via email when an author was invited by an admin and he accepted the invitation using the same email as the one sent by the admin. A different case is when an author receives an admin invitation and he accepts it but using a different email. In that case he receives another email with a confirmation link.

### CODE REVIEW

File: test/oli_web/live/user_live_test.exs [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+    setup [:author_conn]
+
+    test "returns forbidden when accessing the index view", %{conn: conn} do
+      conn = get(conn, @live_view_author_index_route)
+
+      assert response(conn, 403)
+    end
+  end
+
+  describe "index" do
```
Comment (mcass19) [inline test/oli_web/live/user_live_test.exs]: We already add some tests for this route here: https://github.com/wyeworks/oli-torus/blob/master/test/oli_web/live/admin_live_test.exs#L493

Maybe we can update them there to reflect this changes as well?

File: test/oli_web/live/user_actions_test.exs [hunk 2]
```diff
@@ -0,0 +1,96 @@
+defmodule OliWeb.UserActionsTest do
+  use OliWeb.ConnCase
+  use Surface.LiveViewTest
+
+  import Oli.Factory
+
+  alias OliWeb.Users.Actions
+
+  describe "email confirmation" do
```
Comment (mcass19) [inline test/oli_web/live/user_actions_test.exs]: I love testing the components isolated. Just a reminder that we have a test for the authors detail view (decide it by your own if makes sense to add tests there): https://github.com/wyeworks/oli-torus/blob/master/test/oli_web/live/admin_live_test.exs#L579

**Review by mcass19 — COMMENTED** [review]
Great job 🚀  !!! Only minor comments related to tests that you may not have noticed we already have

**Comment by gastonabella** [comment]
@mcass19 thanks for your feedback! I've just pushed the changes

**Review by mcass19 — APPROVED** [review]
👏🏼 👏🏼 👏🏼 👏🏼

**Pull request closed without merging**